src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -0,0 +1,220 @@
+            'table_name' => $options['table_name'] ?? ($query['table_name'] ?? 'messenger_messages'),
+            'queue' => $options['queue'] ?? ($query['queue'] ?? 'default'),
+            'redeliver_timeout' => $options['redeliver_timeout'] ?? ($query['redeliver_timeout'] ?? 3600),
+            'loop_sleep' => $options['loop_sleep'] ?? ($query['loop_sleep'] ?? 200000),
+            'auto_setup' => $options['auto_setup'] ?? ($query['auto_setup'] ?? true),
+        ];
+    }
+
+    public function send(string $body, array $headers, int $delay = 0): void
+    {
+        $availableAt = \DateTime::createFromFormat('U.u', microtime(true) + $delay / 1000);
+
+        $sql = $this->driverConnection->createQueryBuilder()
+            ->insert($this->configuration['table_name'])
+            ->values([
+                'body' => ':body',
+                'headers' => ':headers',
+                'queue' => ':queue',
+                'created_at' => ':create_at',
+                'available_at' => ':available_at',
+            ]);
+        $now = (\DateTime::createFromFormat('U.u', microtime(true)))->format('Y-m-d H:i:s.u');
+        $type = Type::getType(Type::JSON);
+        $this->executeQuery($sql, [
+            ':body' => $body,
+            ':headers' => $type->convertToDatabaseValue($headers, $this->driverConnection->getDatabasePlatform()),
+            ':queue' => $this->configuration['queue'],
+            ':create_at' => $now,
+            ':available_at' => $availableAt->format('Y-m-d H:i:s.u'),
+        ]);
+    }
+
+    public function get(): ?array
+    {
+        $this->driverConnection->beginTransaction();
+        try {
+            $query = $this->driverConnection->createQueryBuilder()
+                ->select('m.*')
+                ->from($this->configuration['table_name'], 'm')
+                ->where('m.delivered_at is null OR m.delivered_at < :redeliver_limit')
+                ->andWhere('m.available_at <= :now')
+                ->andWhere('m.queue = :queue')
+                ->orderBy('available_at', 'ASC')
+                ->setMaxResults(1);
+
+            $redeliverLimit = \DateTime::createFromFormat('U.u', microtime(true) - $this->configuration['redeliver_timeout']);
+            // use SELECT ... FOR UPDATE to lock table
+            $doctrineEnvelope = $this->executeQuery(
+                $query->getSQL().' '.$this->driverConnection->getDatabasePlatform()->getWriteLockSQL(),
+                [
+                    ':now' => (\DateTime::createFromFormat('U.u', microtime(true)))->format('Y-m-d H:i:s.u'),
+                    ':queue' => $this->configuration['queue'],
+                    ':redeliver_limit' => $redeliverLimit->format('Y-m-d H:i:s'),
+                ]
+            )->fetch();
+
+            if (false === $doctrineEnvelope) {
+                $this->driverConnection->commit();
+
+                return null;
+            }
+
+            $type = Type::getType(Type::JSON);
```

Comment:
Do you really need this to be a variable?

Reply:
@vincenttouzet why did you mark this comment as resolved?

Reply:
Woops my bad ... I've made the changes in the send method but not this one ...